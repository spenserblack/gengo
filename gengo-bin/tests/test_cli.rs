use insta::assert_snapshot;
use std::io::{self, Write};

const ROOT: &str = env!("CARGO_MANIFEST_DIR");

struct NullWriter;

impl Write for NullWriter {
    fn write(&mut self, buf: &[u8]) -> io::Result<usize> {
        Ok(buf.len())
    }
    fn flush(&mut self) -> io::Result<()> {
        Ok(())
    }
}

macro_rules! assert_stdout_snapshot {
    ($cli_args:expr $(,)?) => {{
        let cli = gengo_bin::cli::try_new_from($cli_args).unwrap();
        let mut stdout = Vec::new();
        let mut stderr = NullWriter;
        cli.run(&mut stdout, &mut stderr).unwrap();
        let stdout = String::from_utf8(stdout).unwrap();
        assert_snapshot!(stdout);
    }};
}

#[test]
fn test_javascript_repo() {
    assert_stdout_snapshot!(&["gengo", "-r", "test/javascript", "-R", ROOT]);
}

#[test]
#[cfg_attr(windows, ignore)]
fn test_breakdown_javascript_repo() {
    assert_stdout_snapshot!(&["gengo", "-r", "test/javascript", "-R", ROOT, "--breakdown"]);
}

//
// TODO Add test_javascript_repo_windows
