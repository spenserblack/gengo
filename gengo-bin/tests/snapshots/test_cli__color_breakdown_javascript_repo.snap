---
source: gengo-bin/tests/test_cli.rs
expression: stdout
---
[38;2;0;0;0;48;2;240;220;78m 18.42% 28      JavaScript[0m
[38;2;255;255;255;48;2;0;0;0m 40.79% 62      Plain Text[0m
[38;2;255;255;255;48;2;47;116;192m 40.79% 62      TypeScript[0m

[38;2;0;0;0;48;2;240;220;78mJavaScript[0m
  [38;2;0;0;0;48;2;240;220;78mbin.js[0m

[38;2;255;255;255;48;2;0;0;0mPlain Text[0m
  [38;2;255;255;255;48;2;0;0;0mdist/bin.js[0m

[38;2;255;255;255;48;2;47;116;192mTypeScript[0m
  [38;2;255;255;255;48;2;47;116;192msrc/bin.ts[0m


